test/widgets/actions_test.dart

```diff
@@ -52,7 +53,7 @@ void main() {
         processedCount: 11, updatedCount: 3,
         firstProcessedId: null, lastProcessedId: null,
         foundOldest: true, foundNewest: true).toJson());
-      markNarrowAsRead(context, narrow);
+      unawaited(markNarrowAsRead(context, narrow));
       await tester.pump(Duration.zero);
```

Comment:
Huh, odd. I wonder why this test is using `tester.pump` — is that just a substitute for awaiting the future?

What if we replaced both these lines with this?
```dart
      await markNarrowAsRead(context, narrow);
```

(Similarly for the other `markNarrowAsRead` calls in this file.)

Reply:
It's stuck forever, right before the `connection.post` call to "messages/flags/narrow". It's puzzling that it happens to a `FakeApiConnection` though.

Reply:
I guess this is because we call inside `testWidgets`. This issue is not specific to `markNarrowAsRead`. For example, a `store.sendMessage` call stucks as well. The `FakeApiClient` returns a `Future.delayed(response.delay, computation);` that does not complete.

Reply:
Ah, I see. So basically it's running into the problem that `awaitFakeAsync` solves.

It feels odd for these tests to not be awaiting that future — it means they're not ensuring that the requests have actually completed, and that the function has done the work it's going to do when it gets the responses, before they complete.

I'll send a quick separate PR that fixes these.

Reply:
#993